src/main/java/org/jitsi/videobridge/DtlsTransport.java

```diff
@@ -357,19 +358,38 @@ private void installIncomingPacketReader(DatagramSocket socket)
         // enough (it'll only be a bit of the DTLS path) that running it in the
         // IO pool is fine
         TaskPools.IO_POOL.submit(() -> {
-            byte[] bbuf = ByteBufferPool.getBuffer(1500);
-            DatagramPacket p = new DatagramPacket(bbuf, 0, 1500);
+
+            // We need this buffer to be 1500 bytes because we don't know how
+            // big the received packet will be. But we don't want to allocate
+            // large buffers for all packets.
+            byte[] receiveBuf = new byte[1500];
+            DatagramPacket p = new DatagramPacket(receiveBuf, 0, 1500);
+
             while (!closed)
             {
                 try
                 {
                     socket.receive(p);
-                    Packet pkt = new UnparsedPacket(bbuf, 0, p.getLength());
+                    int len = p.getLength();
+                    byte[] buf
+                        = ByteBufferPool.getBuffer(
+                                len +
+                                RtpPacket.BYTES_TO_LEAVE_AT_START_OF_PACKET +
+                                RtpPacket.BYTES_TO_LEAVE_AT_END_OF_PACKET);
+                    System.arraycopy(
```

Comment:
is getting rid of the buffer retrieval/return from/to the pool in need of this change considering we add an arraycopy? 
 we're not `allocating` a big buffer for every one of these packets--they're already allocated, presumably--we're just retrieving and returning them to/from the pool.

Reply:
We were allocating a big buffer, because we need a big buffer with `DatagramSocket.receive` since we don't know how big a packet we'll receive.